lms/static/scripts/frontend_apps/components/LMSFilePicker.js

```diff
@@ -27,13 +27,17 @@ import FileList from './FileList';
 /**
  * @typedef DialogState
  * @prop {'fetching'|'fetched'|'authorizing'|'error'} state
+ * @prop {string} title - Dialog title
+ * @prop {'select'|'authorize'|'authorize_retry'|'retry'|'none'} buttonAction - Action for the right button
```

Comment:
The dialog state contains two more properties: `title` and `buttonAction`.

This prevents from checking several multiple conditions (see below).